js/background.js

```diff
@@ -1826,25 +1825,21 @@
         activeAt = activeAt || Date.now();
       }
       const ourPrimaryKey = window.storage.get('primaryDevicePubKey');
-      const ourDevices = await libloki.storage.getAllDevicePubKeysForPrimaryPubKey(
-        ourPrimaryKey
-      );
-      // TODO: We should probably just *not* send any secondary devices and
-      // just load them all and send FRs when we get the mapping
-      const isOurSecondaryDevice =
-        id !== ourPrimaryKey &&
-        ourDevices &&
-        ourDevices.some(devicePubKey => devicePubKey === id);
-
-      if (isOurSecondaryDevice) {
-        await conversation.setSecondaryStatus(true, ourPrimaryKey);
+      if (ourPrimaryKey) {
+        const secondaryDevices = await libsession.Protocols.MultiDeviceProtocol.getSecondaryDevices(
+          ourPrimaryKey
+        );
+        if (secondaryDevices.some(device => device.key === id)) {
```

Comment:
Just a thought: we should probably have a method for checking if the pubkey is one of our devices. Seems like it would be used a lot.

Reply:
i think this makes sense.
I'll add 2 functions:
`getOurDevices` and `isOurDevice`?

Reply:
Yeah, you are right.